extensions/parrot-translate/src/parrot.tsx

```diff
@@ -0,0 +1,153 @@
+import { COPY_TYPE } from "./consts"
+import { ListActionPanel } from "./components"
+import { Fragment, useEffect, useState } from "react"
+import { Action, ActionPanel, getPreferenceValues, Icon, List } from "@raycast/api"
+import { ILanguageListItem, IPreferences, ITranslateReformatResult, ITranslateResult } from "./types"
+import {
+    requestYoudaoAPI,
+    getItemFromLanguageList,
+    reformatTranslateResult,
+    detectIsUppercaseCopyOrLowerCaseCopy,
+    removeDetectCopyModeSymbol,
+} from "./shared.func"
+
+let fetchResultStateCode = "-1"
+let delayFetchTranslateAPITimer: NodeJS.Timeout
+
+export default function () {
+    const [inputState, updateInputState] = useState<string>()
+    const [isLoadingState, updateLoadingState] = useState<boolean>(false)
+
+    const preferences: IPreferences = getPreferenceValues()
+    const defaultLanguage1 = getItemFromLanguageList(preferences.lang1)
+    const defaultLanguage2 = getItemFromLanguageList(preferences.lang2)
+
+    const [translateTargetLanguage, updateTranslateTargetLanguage] = useState<ILanguageListItem>(defaultLanguage1)
+    const [translateResultState, updateTranslateResultState] = useState<ITranslateReformatResult[]>()
+
+    const [translateFromLanguageState, updateTranslateFromLanguageState] = useState<ILanguageListItem>()
+    const [currentTargetLanguageState, updateCurrentTargetLanguageState] = useState<ILanguageListItem>()
+
+    const [copyModeState, updateCopyModeState] = useState<COPY_TYPE>(COPY_TYPE.Normal)
+
+    useEffect(() => {
+        if (!inputState) return // Prevent when mounted run
+
+        clearTimeout(delayFetchTranslateAPITimer)
+        delayFetchTranslateAPITimer = setTimeout(() => {
+            requestYoudaoAPI(inputState, translateTargetLanguage.languageId).then((res) => {
+                const resData: ITranslateResult = res.data
+
+                const [a, b] = resData.l.split("2") // en2zh
+
+                if (a === b) {
+                    updateTranslateTargetLanguage(a === preferences.lang1 ? defaultLanguage2 : defaultLanguage1)
+                    return
+                }
+
+                updateLoadingState(false)
+                fetchResultStateCode = res.data.errorCode
+                updateTranslateResultState(reformatTranslateResult(resData))
+
+                updateTranslateFromLanguageState(getItemFromLanguageList(a))
+                updateCurrentTargetLanguageState(getItemFromLanguageList(b))
+            })
+        }, 900)
```

Comment:
```suggestion
        requestYoudaoAPI(inputState, translateTargetLanguage.languageId).then((res) => {
            const resData: ITranslateResult = res.data
    
            const [a, b] = resData.l.split("2") // en2zh
    
            if (a === b) {
                updateTranslateTargetLanguage(a === preferences.lang1 ? defaultLanguage2 : defaultLanguage1)
                return
            }
    
            updateLoadingState(false)
            fetchResultStateCode = res.data.errorCode
            updateTranslateResultState(reformatTranslateResult(resData))
    
            updateTranslateFromLanguageState(getItemFromLanguageList(a))
            updateCurrentTargetLanguageState(getItemFromLanguageList(b))
        })
```
With throttle on the list, this shouldn't be necessary